src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -529,10 +529,12 @@ void MacroAssembler::resolve_jobject(Register value, Register thread, Register t
 }
 
 void MacroAssembler::stop(const char* msg) {
+  // these shall generate fixed-length instruction sequences
```

Comment:
Looks reasonable to me. (Not a JDK Reviewer)
Suggestion for the code comment:

@@ -531,6 +531,9 @@ void MacroAssembler::resolve_jobject(Register value, Register thread, Register t
 void MacroAssembler::stop(const char* msg) {
   address ip = pc();
   pusha();
+  // The length of the instruction sequence emitted should be independent
+  // of the values of msg and ip so that size of mach node for scratch emit
+  // and normal emit.
   li(c_rarg0, (uintptr_t)(address)msg);
   li(c_rarg1, (uintptr_t)(address)ip);
   mv(c_rarg2, sp);

Reply:
Thanks for reviewing this patch and the polish of comments, Felix. It looks more precise than before.